lib/matplotlib/cbook.py

```diff
@@ -286,7 +286,7 @@ def process(self, s, *args, **kwargs):
         """
         if self._signals is not None:
             _api.check_in_list(self._signals, signal=s)
-        for cid, ref in list(self.callbacks.get(s, {}).items()):
+        for ref in list(self.callbacks.get(s, {}).values()):
```

Comment:
```suggestion
        for ref in self.callbacks.get(s, {}).values():
```
?

Or is there an obvious reason to make a list first? (There may be, but since we are talking micro-optimizations...)

Reply:
If there is a reason, please make a comment if possible.

Reply:
I think we used to clean out expired callbacks here?  I have a vague memory of this being required to avoid mutating the dict while iterating over it...but clearly that is not happening in this code now.

Reply:
Yeah, I couldn't see any mutations either.

Reply:
It looks like 5257c4f2326286dfa5bfa244613c9d31c8cf0b99 removed the mutation in the loop.

Reply:
A possible mutation of `self.callbacks` in the loop was the reason I did not remove the conversion to list. In theory the `func` is an arbitrary method, but maybe this is no concern in practice.

Let me know if you want me to remove the list conversion.

Reply:
Likely this is necessary in case a callback removed itself (for example), in which case the list will be mutated. (I think.)

Reply:
I assume unpacking an iterable into a list using `[*foo]` is faster than calling the `list` constructor: `list(foo)`? Something to do with CPython implementation details? Not likely to save much time, but then again: micro-optimisations.

Another case against this is that unpacking is may be less readable.